examples/RNOneSignal/App.js

```diff
@@ -8,484 +8,700 @@
+//        OneSignal.removeEventListener('subscription', this.onSubscriptionChange);
+//        OneSignal.removeEventListener('permission', this.onPermissionChange);
+        OneSignal.removeEventListener('emailSubscription', this.onEmailSubscriptionChange);
+        OneSignal.removeEventListener('inAppMessageClicked', this.onInAppMessageClicked);
+    }
+
+    /**
+     Validate email method
+     */
+    validateEmail(email) {
+        var re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+        return re.test(String(email).toLowerCase());
+    }
+
+    /**
+     In-App Message public method examples
+     */
+    oneSignalInAppMessagingExamples() {
+        // Add a single trigger with a value associated with it
+        OneSignal.addTrigger('trigger1', 'one');
+
+        // Get trigger value for the key
+        OneSignal.getTriggerValueForKey('trigger1')
+            .then(response => {
+                console.log('trigger1 value: ' + response);
+            })
+            .catch(e => {
+                console.error(e);
+            });
+
+        // Create a set of triggers in a map and add them all at once
+        var triggers = {
+            trigger2: '2',
+            trigger3: true,
+        };
+        OneSignal.addTriggers(triggers);
+
+        // Get trigger value for the key
+        OneSignal.getTriggerValueForKey('trigger3')
+            .then(response => {
+                console.log('trigger3 value: ' + response);
+            })
+            .catch(e => {
+                console.error(e);
+            });
+
+        // Create an array of keys to remove triggers for
+        var removeTriggers = ['trigger1', 'trigger2'];
+        OneSignal.removeTriggersForKeys(removeTriggers);
+    }
+
+    /**
+     Outcomes public method examples
+     */
+    oneSignalOutcomeEventsExamples() {
+        // Send an outcome event with and without a callback
+        OneSignal.sendOutcome('normal_1');
+        OneSignal.sendOutcome('normal_2', response => {
+              console.log('Normal outcome sent successfully!');
+              console.log(response);
+        });
+
+        // Send a unique outcome event with and without a callback
+        OneSignal.sendUniqueOutcome('unique_1');
+        OneSignal.sendUniqueOutcome('unique_2', response => {
+              console.log('Unique outcome sent successfully!');
+              console.log(response);
+        });
+
+        // Send an outcome event with and without a callback
+        OneSignal.sendOutcomeWithValue('value_1', 9.99);
+        OneSignal.sendOutcomeWithValue('value_2', 5, response => {
+              console.log('Outcome with value sent successfully!');
+              console.log(response);
+        });
+    }
+
+    /**
+     When a notification is received this will fire
+     */
+    onNotificationReceived(notification) {
```

Comment:
this can be transformed to an arrow function 
onNotificationReceived = (notification) -> {
....
}

with this you can delete the binding

Reply:
if you are curious https://medium.com/@joespinelli_6190/using-arrow-functions-to-avoid-binding-this-in-react-5d7402eec64

Reply:
Replaced all binds. thank you so much, much cleaner